src/hooks/use-pull-push-timestamps.ts

```diff
@@ -1,92 +1,97 @@
-			pull?: number;
-			push?: number;
-		};
+	const fetchConnectedSite = async ( siteId: string, connectedSiteId: number ) => {
+		const connectedSites = await getIpcApi().getConnectedWpcomSites( siteId );
+		return connectedSites.find( ( site ) => site.id === connectedSiteId );
 	};
-}
 
-export function usePullPushTimestamps() {
-	const [ timestamps, setTimestamps ] = useState< SyncTimestamps >( () => {
-		try {
-			const stored = localStorage.getItem( SYNC_TIMESTAMPS_STORAGE_KEY );
-			return stored ? JSON.parse( stored ) : {};
-		} catch ( e ) {
-			console.error( 'Failed to parse sync timestamps:', e );
-			return {};
-		}
-	} );
+	const getLastSyncTimeWithType = useCallback(
+		async ( siteId: string, connectedSiteId: number, type: 'pull' | 'push' ): Promise< string > => {
+			try {
+				const site = await fetchConnectedSite( siteId, connectedSiteId );
+				if ( ! site ) return __( 'Site not found.' );
 
-	useEffect( () => {
-		try {
-			localStorage.setItem( SYNC_TIMESTAMPS_STORAGE_KEY, JSON.stringify( timestamps ) );
-		} catch ( e ) {
-			console.error( 'Failed to save sync timestamps:', e );
-		}
-	}, [ timestamps ] );
+				const timestamp = type === 'pull' ? site.lastPullTimestamp : site.lastPushTimestamp;
 
-	const updateTimestamp = useCallback(
-		( localSiteId: string, connectedSiteId: number, type: 'pull' | 'push' ) => {
-			setTimestamps( ( prev ) => {
-				const newTimestamps = { ...prev };
-				if ( ! newTimestamps[ localSiteId ] ) {
-					newTimestamps[ localSiteId ] = {};
-				}
-				if ( ! newTimestamps[ localSiteId ][ connectedSiteId ] ) {
-					newTimestamps[ localSiteId ][ connectedSiteId ] = {};
+				if ( timestamp === null ) {
+					return type === 'pull'
+						? __( 'You have not pulled this site yet.' )
+						: __( 'You have not pushed this site yet.' );
 				}
-				newTimestamps[ localSiteId ][ connectedSiteId ][ type ] = Date.now();
-				return newTimestamps;
-			} );
+
+				return sprintf(
+					type === 'pull'
+						? __( 'You pulled this site %s ago.' )
+						: __( 'You pushed this site %s ago.' ),
+					formatDistanceToNow( new Date( timestamp ) )
+				);
+			} catch ( error ) {
+				console.error( 'Failed to get sync time:', error );
+				return __( 'Unable to determine sync time.' );
+			}
 		},
 		[]
 	);
 
-	const getLastSyncTimeWithType = useCallback(
-		( localSiteId: string, connectedSiteId: number, type: 'pull' | 'push' ): string => {
-			const localSiteTimestamps = timestamps[ localSiteId ] || {};
-			const siteTimestamps = localSiteTimestamps[ connectedSiteId ] || {};
-			const timestamp = siteTimestamps[ type ];
+	const updateTimestamp = useCallback(
+		async ( siteId: string, connectedSiteId: number, type: 'pull' | 'push' ) => {
+			try {
+				const site = await fetchConnectedSite( siteId, connectedSiteId );
+				if ( ! site ) return;
+
+				const updatedSite = {
+					...site,
+					[ type === 'pull' ? 'lastPullTimestamp' : 'lastPushTimestamp' ]: new Date().toISOString(),
```

Comment:
Not a strong opinion, but why saving it as a String?
If it were a database, I would save it as a number. Like this `new Date().getTime()`

Reply:
Yeah I initially had that in https://github.com/Automattic/studio/pull/714/commits/177dd5df11597cdf905e3cd86371b47b2fba212a but doing some research, I saw that `to.ISOString` was a common way to save dates that would format it as `YYYY-MM-DDZHH:mm:ss` so I went with that approach. I think it makes it look more straightforward that it is a date. It is not a strong preference though so happy to change that 👍